### ISSUE TITLE
Program crashes at random when calling append_mesh

### ISSUE THREAD

**Issue by ygrimault** [libigl/libigl#1043]
When I try to increase the size of `data_list` with `append_mesh()`, my program *sometimes* crashes at the point where it calls `data_list.emplace_back()`. It's not consistent, sometimes I can load 3, 4 complex meshes without a problem, but sometimes it crashes when the only mesh I've loaded is a square (4 vertices, 2 faces).

Have you already encountered this problem ? The main problem I have is that it's not consistent, it may or may not happen (but mostly happens, I'd say 70% of the time maybe).

My program is a simple loader with an ImGUI menu like example 106, so I mainly load objects using `open_dialog_load_mesh()`, but I also create square meshes manually, calling directly `append_mesh()` before clearing `data()` and setting the new mesh. However the crash occurs with both methods, and at the point I described above.

#### Check all that apply (change to `[x]`)
- [x] Windows
- [ ] Mac OS X
- [ ] Linux

**Comment by jdumas** [comment]
Are you using the `dev` branch, or the `master` branch? Are you compiling in 32bits or 64bits on Windows? If the former, could this be related to #1029?

**Comment by ygrimault** [comment]
I'm using the `master` branch, and compiling in 64bits on Windows.

This particular issue doesn't seem to be directly related to the issue you linked, as it doesn't say anything else.
HOWEVER, I do have the issue you linked from time to time, at the start of the program. The exact issue is:
```
Assertion failed: (reinterpret_cast<size_t>(array) & 0xf) == 0 && "this assertion is explained here: " "http://eigen.tuxfamily.org/dox-devel/group__TopicUnalignedArrayAssert.html" "
**** READ THIS WEB PAGE !!! ****", file c:\...\libigl\external\eigen\eigen\src/Core/DenseStorage.h, line 86
```
but I haven't looked where it comes from in the code (I don't have any debugger as of now in QtCreator).

**Comment by jdumas** [comment]
Well, please do try the `dev` branch then, and let us know if the issue ever pops up again.

**Comment by ygrimault** [comment]
Thanks for the tip, it does seem to have solved both my problems (I ran the program several times in a row without the problem occuring)!
However, I needed to modify a bit `libigl\include\igl\opengl\glfw\imgui\ImGuiTraits.h` as I was getting the following error: `expression did not evaluate to a constant`. I guess that it's due to MSCV, but there's an easy fix: just use what is on the `master` branch, ie change all `format []` to `* format`.
